indy_node/server/restarter.py

```diff
@@ -0,0 +1,288 @@
+import os
+from collections import deque
+from datetime import datetime, timedelta
+from functools import partial
+from typing import Tuple, Union, Optional, Callable, Dict
+
+import dateutil.parser
+import dateutil.tz
+
+from indy_node.server.node_maintainer import NodeMaintainer
+from indy_node.server.restart_log import RestartLog
+from stp_core.common.log import getlogger
+from plenum.common.constants import TXN_TYPE, VERSION, DATA, IDENTIFIER
+from plenum.common.types import f
+from plenum.server.has_action_queue import HasActionQueue
+from indy_common.constants import ACTION, POOL_RESTART, START, SCHEDULE, \
+    CANCEL, JUSTIFICATION, TIMEOUT, REINSTALL, IN_PROGRESS, FORCE
+from plenum.server import notifier_plugin_manager
+from ledger.util import F
+import asyncio
+
+logger = getlogger()
+
+
+class Restarter(NodeMaintainer):
+
+    def _defaultLog(self, dataDir, config):
+        log = os.path.join(dataDir, config.restartLogFile)
+        return RestartLog(filePath=log)
+
+    def _is_action_started(self):
+        if not self.lastActionEventInfo:
+            logger.debug('Node {} has no restart events'
+                         .format(self.nodeName))
+            return False
+
+        (event_type, when) = self.lastActionEventInfo
+
+        if event_type != RestartLog.STARTED:
+            logger.debug(
+                'Restart for node {} was not scheduled. Last event is {}:{}:{}'.format(
+                    self.nodeName, event_type, when))
+            return False
+
+        return True
+
+    def _update_action_log_for_started_action(self):
+        (event_type, when) = self.lastActionEventInfo
+
+        if not self.didLastExecutedRestartSucceeded:
```

Comment:
Where this field is assigned?

Reply:
Nowhere. Remove this condition or think through the solution if the restart is unsuccessful.